gr-digital/lib/costas_loop_cc_impl.cc

```diff
@@ -50,43 +50,36 @@ costas_loop_cc_impl::costas_loop_cc_impl(float loop_bw, int order, bool use_snr)
       d_order(order),
       d_error(0),
       d_noise(1.0),
-      d_phase_detector(NULL)
+      d_phase_detector(choose_phase_detector(order, use_snr))
 {
-    // Set up the phase detector to use based on the constellation order
-    switch (d_order) {
-    case 2:
-        if (use_snr)
-            d_phase_detector = &costas_loop_cc_impl::phase_detector_snr_2;
-        else
-            d_phase_detector = &costas_loop_cc_impl::phase_detector_2;
-        break;
-
-    case 4:
-        if (use_snr)
-            d_phase_detector = &costas_loop_cc_impl::phase_detector_snr_4;
-        else
-            d_phase_detector = &costas_loop_cc_impl::phase_detector_4;
-        break;
-
-    case 8:
-        if (use_snr)
-            d_phase_detector = &costas_loop_cc_impl::phase_detector_snr_8;
-        else
-            d_phase_detector = &costas_loop_cc_impl::phase_detector_8;
-        break;
-
-    default:
-        throw std::invalid_argument("order must be 2, 4, or 8");
-        break;
-    }
-
     message_port_register_in(pmt::mp("noise"));
     set_msg_handler(pmt::mp("noise"),
                     boost::bind(&costas_loop_cc_impl::handle_set_noise, this, _1));
 }
 
 costas_loop_cc_impl::~costas_loop_cc_impl() {}
 
+costas_loop_cc_impl::d_phase_detector_t costas_loop_cc_impl::choose_phase_detector(int order, bool use_snr)
+{
+  switch (order) {
+  case 2:
+    if (use_snr)
```

Comment:
If you're already touching this, please add {} so clang-tidy won't freak out once we enable that.

Reply:
Oooh, braces around every `if`? Great!